Samples/2.3/HealthCheckSample/src/Silo/ClusterHealthCheck.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Orleans;
+using Orleans.Runtime;
+
+namespace Silo
+{
+    public class ClusterHealthCheck : IHealthCheck
+    {
+        private readonly IClusterClient client;
+
+        public ClusterHealthCheck(IClusterClient client)
+        {
+            this.client = client;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            var manager = client.GetGrain<IManagementGrain>(0);
+            try
+            {
+                var hosts = await manager.GetHosts();
+                var count = hosts.Values.Where(_ => _.IsUnavailable()).Count();
```

Comment:
i suggest not to use a discard naming (`_`) for a non discarded property (since its used)

```suggestion
                var count = hosts.Values.Where(x => x.IsUnavailable()).Count();
```